app/models/form_handler.rb

```diff
@@ -47,7 +47,7 @@ def lettings_forms
         form = Form.new(form_path)
 
         form_to_set = form_name_from_start_year(form.start_date.year, "lettings")
-        forms[form_to_set] = form if forms[form_to_set].blank?
+        forms[form_to_set] = form if form_to_set && forms[form_to_set].blank?
       end
```

Comment:
@kosiakkatrina 
`Dir.glob("#{directory}/*.json") => ["config/2021_2022.json", "config/2022_2023.json"]`
for 21/22 we will call `form_name_from_start_year(2021, "lettings")`
in form_name_from_start_year below line 71 you will see that will return
`form_mappings[2023 - 2021]` => `form_mappings[2]` => `nil`
so back here on line 50 we will call `forms[nil] = bla bla`

This is a problem because when we create a log, we go and get the relevant form. we use the saledate/startdate to get the right form name and call get_form (above line 23).
if there is no startdate/saledate, then currently we call `get_form(nil)` which returns `nil` and we then default in the log `:form` method to the current form for that type:
`FormHandler.instance.get_form(form_name) || FormHandler.instance.current_lettings_form`
In the crossover period, if we allow `FormHandler.instance.forms` to have a nil key in it, then until saledate/startdate is set we default to the 21/22 lettings form. This crashes the site for creating new sales logs entirely for obvious reasons and the lettings log section doesn't allow the setup subsection link to function.

I think that we don't want to be adding things to our forms hash under the key `nil`
We may also want to make our `:form` methods in `SalesLog` and `LettingsLog` more robust to deal with this case? I didn't immediately feel that was necessary as long as the forms hash is kept tidy.

Reply:
Okay, yes, this makes sense!
I've got a different question tho, in the crossover period, the form should default to 23/24, because that would be the `current_form` for that time. Is that not what's happening?

Reply:
that will be what will happen with the change that I have made because 
`FormHandler.instance.get_form(form_name) || FormHandler.instance.current_lettings_form`
becomes `FormHandler.instance.get_form(nil) || FormHandler.instance.current_lettings_form'
becomes `nil || FormHandler.instance.current_lettings_form`
but without the change (ie `FormHandler.instance.forms => {nil => <#Form 21-22>, "current_sales" => <#Form bla>, etc}`:
`FormHandler.instance.get_form(form_name) || FormHandler.instance.current_lettings_form`
becomes `FormHandler.instance.get_form(nil) || FormHandler.instance.current_lettings_form'
becomes `<#Form 21-22> || FormHandler.instance.current_lettings_form`
and this makes crashy crashy